source/sn_coap_builder.c

```diff
@@ -144,16 +142,26 @@ int16_t sn_coap_builder_2(uint8_t *dst_packet_data_ptr, const sn_coap_hdr_s *src
         /* * * * * * * * * * * * * * * * * * */
         sn_coap_builder_payload_build(&dst_packet_data_ptr, src_coap_msg_ptr);
     }
+
+    /* Shout as much as we can about overflow - if we exceed this, may have overrun user's buffer */
+    if (dst_packet_data_ptr - base_packet_data_ptr > dst_byte_count_to_be_built) {
```

Comment:
Maybe we could add `tr_error` here to spot it from the traces?

Reply:
I did want an assert, but the library doesn't already use them, so would be a new porting dependency. If this happens, we've probably overrun memory already - we don't get an "output buffer length" parameter.

A `tr_error` would make sense as a substitute, yes.